Training/Training/TestCase.c

```diff
@@ -0,0 +1,158 @@
+// DecimalConversion.c
+// C program to check the test cases of Decimal conversion
+// ------------------------------------------------------------------------------------------------
+
+#include <stdio.h>
+#include <ctype.h>
+#include <string.h>
+#include "DecimalConversion.h"
+
+#define ANSI_RESET_ALL "\x1b[0m"
+#define ANSI_COLOR_RED "\x1b[31m"
+#define ANSI_COLOR_GREEN "\x1b[32m"
+#define ANSI_COLOR_BLUE "\x1b[36m"
+#define ANSI_COLOR_MAGENTA "\x1b[35m"
+
+//integer range
+#define INT_MAX 2147483648
+#define INT_MIN -2147483649ll
+
+/// <summary>Function to get only integer input from user</summary>
+long long int GetIntegerOnly ();
+/// <summary>Function to check binary values</summary>
+void TestDecToBin (int input[], char* outputBin[], int arrLength);
+/// <summary>Function to check hexadecimal values</summary>
+void TestDecToHex (int input[], char* outputHex[], int arrLength);
+
+
+void main () {
+//test cases for Binary conversion
+   int input1[] = { 0,651,-545,999,-4875,15756,-4654,-1,-2147483647 };
+   int length1 = sizeof (input1) / sizeof (input1[0]);
+   char* output1[] = {
+      "00000000000000000000000000000000",
+      "00000000000000000000001010001011",
+      "11111111111111111111110111011111",
+      "00000000000000000000001111100111",
+      "11111111111111111110110011110101",
+      "00000000000000000011110110001100",
+      "11111111111111111110110111010010",
+      "11111111111111111111111111111111",
+      "10000000000000000000000000000001" };
+   TestDecToBin (input1, output1, length1);
+   char* output2[] = {
+        "00000000",
+        "0000028B",
+        "FFFFFDDF",
+        "000003E7",
+        "FFFFECF5",
+        "00003D8C",
+        "FFFFEDD2",
+        "FFFFFFFF",
+        "80000001" };
+   TestDecToHex (input1, output2, length1);
+   printf ("\n");
+//test cases for Hexadecimal conversion
+   int input2[] = { 1,255,1023,16384,2147483647,-1,-5,-2147483647,-2147483648ll };
+   int length2 = sizeof (input2) / sizeof (input2[0]);
+   char* output3[] = {
+      "00000000000000000000000000000001",
+      "00000000000000000000000011111111",
+      "00000000000000000000001111111111",
+      "00000000000000000100000000000000",
+      "01111111111111111111111111111111",
+      "11111111111111111111111111111111",
+      "11111111111111111111111111111011",
+      "10000000000000000000000000000001",
+      "10000000000000000000000000000000" };
+   TestDecToBin (input2, output3, length2);
+   char* output4[] = {
+      "00000001",
+      "000000FF",
+      "000003FF",
+      "00004000",
+      "7FFFFFFF",
+      "FFFFFFFF",
+      "FFFFFFFB",
+      "80000001",
+      "80000000" };
+   TestDecToHex (input2, output4, length2);
+//User input validation
+   while (1) {
+      long long int dec = 0;
+      dec = GetIntegerOnly ();
+      if (dec != INT_MIN) {
+         char* binNum = DecToBin ((unsigned int)dec);
+         printf ("Binary value(0b): "ANSI_COLOR_BLUE"%s"ANSI_RESET_ALL, binNum);

```

Comment:
![image](https://github.com/user-attachments/assets/e1bd0167-b119-4b0d-9514-5222c93fbea3)
0b and 0x should be at the beginning of your result following standard notation